tests/cli/test_sanitize_yaml_and_save_datasource.py

```diff
@@ -42,24 +42,23 @@ def test_sanitize_yaml_and_save_datasource_works_without_credentials(
     context = empty_data_context
     yaml_snippet = """
 name: my_datasource
-class_name: SimpleSqlalchemyDatasource
-introspection:
-  whole_table:
-    data_asset_name_suffix: __whole_table
-connection_string: sqlite://"""
+class_name: Datasource
+execution_engine:
+  class_name: PandasExecutionEngine
+"""
```

Comment:
`non-blocking`

Why did these BDS examples need to change?

Reply:
Good catch - meant to revert. Initial approach to `DatasourceDict.set_datasource` attempted to instantiate and return `BaseDatasource` obj from config for BDS, but this broke a bunch of tests (these changes were first few attempts to fix). Since behavior of returning a Datasource from this method is new and only relevant to FDS, decided to just return None for BDS vs. instantiating, instead of wasting time refactoring tests.

Addressed in https://github.com/great-expectations/great_expectations/pull/8773/commits/e51aa04faaf82e6c32e8635e0f9a42f36c54e2a2.